test/boost/cql_query_test.cc

```diff
@@ -5804,6 +5804,20 @@ SEASTAR_TEST_CASE(test_sending_tablet_info_unprepared_select) {
     }, tablet_cql_test_config());
 }
 
+// Reproduces #20768
+SEASTAR_TEST_CASE(test_alter_keyspace_updates_in_memory_objects_with_data_from_system_schema_scylla_keyspaces) {
+        return do_with_cql_env([] (cql_test_env& e) {
+            return e.execute_cql("create keyspace ks_tablet with replication = { 'class': 'NetworkTopologyStrategy', 'replication_factor': 1 } and tablets = {'initial': 1 }").discard_result().then([&e] {
+                return e.execute_cql("alter keyspace ks_tablet with tablets = { 'initial': 2 };").discard_result().then([&e] {
```

Comment:
Nit: You could use do_with_cql_env_thread(), which allows you to use .get() on futures instead of then(). The code would be easier to read because it will have no nested lambdas. 

Or you could use coroutines, but threads are easier to debug (full stack trace) so preferred in tests.

Reply:
I used `do_with_cql_env_thread`, thanks for the tip.